custom/icds_reports/management/commands/create_citus_child_tables.py

```diff
@@ -0,0 +1,138 @@
+    AGG_CCS_RECORD_DELIVERY_TABLE,
+    AGG_CCS_RECORD_BP_TABLE,
+    'daily_attendance',
+    'child_health_monthly',
+    'ccs_record_monthly',
+    AGG_INFRASTRUCTURE_TABLE,
+]
+
+"""
+- table get's created with normal django migrations
+- dump and load direct
+"""
+plain_tables = [
+    'icds_months',
+    'icds_months_local',
+    'awc_location',
+    'awc_location_local',
+    AggregateInactiveAWW._meta.db_table,
+    IcdsFile._meta.db_table,
+    ICDSAuditEntryRecord._meta.db_table,
+
+]
+
+
+def get_parent_child_mapping(connection):
+    res = connection.execute("""
+        SELECT c.relname AS child, p.relname as parent
+        FROM pg_inherits JOIN pg_class AS c ON (inhrelid=c.oid)
+        JOIN pg_class as p ON (inhparent=p.oid);
+    """)
+
+    mapping = defaultdict(list)
+    for row in res:
+        mapping[row.parent].append(row.child)
+    return mapping
+
+
+def table_exists(connection, table_name):
+    res = connection.execute("select 1 from pg_tables where tablename = %s", table_name)
+    return bool(list(res))
+
+
+class Command(BaseCommand):
+    help = "Compare tables in PG and CitusDB and create missing child tables in Citus."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-s', '--source-db-alias', default='icds-ucr',
+            help='Django alias for source database'
+        )
+        parser.add_argument(
+            '-t', '--target-db-alias', default='icds-ucr-citus',
+            help='Django alias for target database'
+        )
+        parser.add_argument(
+            '--dry-run',
+            action='store_true',
+            help='Only output the commands.',
+        )
+
+    def handle(self, source_db_alias, target_db_alias, **options):
+        self.dry_run = options['dry_run']
+
+        with connection_manager.get_engine(source_db_alias).begin() as conn:
+            self.parent_child_mapping = get_parent_child_mapping(conn)
+
+        source_engine = connection_manager.get_engine(source_db_alias)
+        target_engine = connection_manager.get_engine(target_db_alias)
+        with source_engine.begin() as source_conn, target_engine.begin() as target_conn:
+            for table in keep_child_tables:
+                self.create_child_tables(source_conn, target_conn, table)
+
+    def create_child_tables(self, source_connection, target_connection, table_name):
+        for child_table in self.parent_child_mapping[table_name]:
+            if not table_exists(target_connection, child_table):
+                insp = sqlinspect(source_connection)
+                # only interested in child table constraint and not inherited
+                constraints = [
+                    constraint for constraint in insp.get_check_constraints(child_table)
+                    if constraint['name'].startswith(child_table)
+                ]
+                sql = """
+                    CREATE TABLE IF NOT EXISTS "{tablename}" ({constraint}) INHERITS ("{parent_tablename}")
```

Comment:
it looks like with `INHERITS` none of the indices or constraints are maintained
> All check constraints and not-null constraints on a parent table are automatically inherited by its children, unless explicitly specified otherwise with NO INHERIT clauses. Other types of constraints (unique, primary key, and foreign key constraints) are not inherited.

I know we can get those using `LIKE`, although the tables are decoupled at creation that way. Are we planning on copying indices or unique constraints some other way?